lib/project_types/theme/forms/connect.rb

```diff
@@ -0,0 +1,34 @@
+module Theme
+  module Forms
+    class Connect < ShopifyCli::Form
+      attr_accessor :name
+      flag_arguments :themeid, :password, :store, :env
+
+      def ask
+        self.store ||= CLI::UI::Prompt.ask(ctx.message('theme.forms.ask_store'), allow_empty: false)
+        ctx.puts(ctx.message('theme.forms.connect.private_app', store))
+        self.password ||= CLI::UI::Prompt.ask(ctx.message('theme.forms.ask_password'), allow_empty: false)
+
+        errors = []
+        errors << "store" if store.strip.empty?
+        errors << "password" if password.strip.empty?
+        ctx.abort(ctx.message('theme.forms.errors', errors.join(", ").capitalize)) unless errors.empty?
+
+        self.themeid, self.name = ask_theme(store: store, password: password, themeid: themeid)
```

Comment:
Nitpick: since this is a private method that's only called in a single place, we may want to consider just inlining it into the method.

Reply:
In my mind, it's more organized this way, particularly because we return the ID and name of the theme in one go. Is that a valid reason to extract this into a separate method?

Reply:
Yeah, like I said I'm happy to merge what is there now, just thought I'd raise the point. Readability is definitely a valid reason for extracting a method.